ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -118,73 +119,102 @@ class Endpoints(
     }
   }
 
+  def withJwtPayloadLoggingContext[A, B](jwtPayload: JwtPayload)(
+      fn: LoggingContextOf[JwtPayload with InstanceUUID with RequestID] => A
+  )(implicit lc: LoggingContextOf[InstanceUUID with RequestID]): A =
+    withEnrichedLoggingContext(
+      LoggingContextOf.label[JwtPayload],
+      Map(
+        "ledger_id" -> jwtPayload.ledgerId.toString,
+        "act_as" -> jwtPayload.actAs.toString,
+        "application_id" -> jwtPayload.applicationId.toString,
+        "read_as" -> jwtPayload.readAs.toString,
+      ),
+    ).run(lc => fn(lc))
+
+  // I think this is not the best name.
+  // Naming things is hard :/
+  def handleCommand[T[_]](req: HttpRequest)(
+      fn: (
+          Jwt,
+          JwtWritePayload,
+          JsValue,
+      ) => LoggingContextOf[JwtWritePayload with InstanceUUID with RequestID] => ET[T[ApiValue]]
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
```

Comment:
Likewise.

Reply:
Just noticed that this doesn't work because the other functions require the specific logging context within that function already. Going to revert the changes.

Reply:
That's what the `label` is for, if you're referring to ambiguity.